tests/test_compute_moments.py

```diff
@@ -0,0 +1,79 @@
+import numpy
+import pytest
+from anisoap_rust_lib import compute_moments
+from anisoap.utils.moment_generator import compute_moments_inefficient_implementation
+import timeit
+
+
+TEST_CEN = [
+    numpy.array([1.0, 2.0, 3.0]),
+    numpy.array([0.3, 0.7, 0.1]),
+    numpy.array([0.0, 1.0, 0.0]),
+    numpy.array([1.3, 5.2, 2.7]),
+]
+TEST_MAT = [
+    numpy.array([[0.81, 0.92, 0.93], [0.92, 0.29, 0.66], [0.93, 0.66, 0.13]]),
+    numpy.array([[0.25, 0.54, 0.41], [0.54, 0.79, 0.68], [0.41, 0.68, 0.25]]),
+    numpy.array([[0.95, 0.91, 0.22], [0.91, 0.92, 0.23], [0.22, 0.23, 0.70]]),
+]
+TEST_LMAX = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+
+class TestComputeMoments:
+    @pytest.mark.parametrize("cen", TEST_CEN)
+    @pytest.mark.parametrize("mat", TEST_MAT)
+    @pytest.mark.parametrize("lmax", TEST_LMAX)
+    def test_fixed_input(self, cen, mat, lmax):
+        res_ori: numpy.ndarray[numpy.float64] = (
+            compute_moments_inefficient_implementation(mat, cen, lmax)
+        )
+        res_ffi: numpy.ndarray[numpy.float64] = compute_moments(mat, cen, lmax)
+        assert numpy.allclose(res_ffi, res_ori, 1e-8)
+
+    @pytest.mark.parametrize("cen", TEST_CEN)
+    @pytest.mark.parametrize("mat", TEST_MAT)
+    @pytest.mark.parametrize("lmax", TEST_LMAX)
+    def test_speedup(self, cen, mat, lmax):
+        # Prime the compute_moments rust function, since initial startup can be long.
+        compute_moments(mat, cen, lmax)
+        start = timeit.default_timer()
+        for i in range(15):
+            compute_moments_inefficient_implementation(mat, cen, lmax)
+        time_ori = timeit.default_timer() - start
+
+        start = timeit.default_timer()
+        for i in range(15):
+            compute_moments(mat, cen, lmax)
+        time_ffi = timeit.default_timer() - start
+
+        assert time_ffi < time_ori
```

Comment:
Throwing this in a loop seems overkill. 
```suggestion
        # Prime the compute_moments rust function, since initial startup can be long.
        compute_moments(mat, cen, lmax)

        start = timeit.default_timer()
        compute_moments_inefficient_implementation(mat, cen, lmax)
        time_ori = timeit.default_timer() - start

        start = timeit.default_timer()
        compute_moments(mat, cen, lmax)
        time_ffi = timeit.default_timer() - start

        assert time_ffi < time_ori
```

Reply:
imo, looping is good to reduce variability. The total time to run it 15 times is 1e-3 s for python and 1e-5 s for rust so it's not at all a time sink.

Reply:
15 is overkill, maybe look 3 times MAX. But even then, the timers include 15X the computation time, so the discrepancy will be much bigger than for each run, which is misleading. Feel free to structure it as I suggested and place the entire block in a loop that goes a few times.